src/ngraph/runtime/reference/concat.hpp

```diff
@@ -32,12 +32,11 @@ namespace ngraph
                         T* out,
                         const std::vector<Shape>& in_shapes,
                         const Shape& out_shape,
-                        size_t concatenation_axis)
+                        int64_t concatenation_axis)
             {
```

Comment:
You need to handle negative concatenation_axis.
concatenation_axis = concatenation_axis < 0 ? static_cast<int64_t>(in_shapes.size()) + concatenation_axis : concatenation_axis;